ui/components/app/assets/hooks/useMultichainAssets.tsx

```diff
@@ -0,0 +1,77 @@
+import { useSelector } from 'react-redux';
+import { CaipAssetId, Hex } from '@metamask/utils';
+import { getMultichainSelectedAccountCachedBalance } from '../../../../selectors/multichain';
+import {
+  getMultiChainAssets,
+  getSelectedInternalAccount,
+} from '../../../../selectors';
+import {
+  TranslateFunction,
+  networkTitleOverrides,
+} from '../util/networkTitleOverrides';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { formatWithThreshold } from '../util/formatWithThreshold';
+import { getIntlLocale } from '../../../../ducks/locale/locale';
+import { getCurrentCurrency } from '../../../../ducks/metamask/metamask';
+import {
+  MULTICHAIN_PROVIDER_CONFIGS,
+  MultichainNetworks,
+} from '../../../../../shared/constants/multichain/networks';
+import { TokenWithFiatAmount } from '../types';
+
+const useMultiChainAssets = () => {
+  const t = useI18nContext();
+  const locale = useSelector(getIntlLocale);
+  const selectedAccount = useSelector(getSelectedInternalAccount);
+  const currentCurrency = useSelector(getCurrentCurrency);
+
+  const multichainAssets = useSelector((state) =>
+    getMultiChainAssets(state, selectedAccount),
+  );
+
+  // the following condition is needed to satisfy e2e check-balance.spec.ts
+  // this is because the new multichain data is not being mocked within the withSolanaAccountSnap test fixture
+  // balances render as expected without this condition during local testing
```

Comment:
Could we introduce a mock for this now rather than hard patching something in the hook ?

Reply:
@GuillaumeRx I would personally prefer to add this in as a follow up. Main reason being that the changes in this PR (namely the selector) are blocking two other teams cc @FrederikBolding @micaelae 

We are planning to discuss this further with @javiergarciavera to create a ticket for this

Reply:
I'm from one of the teams waiting for this (snaps), I'm well aware :sweat_smile: , I agree this shouldn't block the PR but if this could be resolved sooner than later it would be great.

Reply:
We've opened a thread with QA and will make sure we prioritize a quick follow up here.